fluent-bit-to-loki/pkg/lokiplugin/utils.go

```diff
@@ -87,6 +88,31 @@ func autoLabels(records map[string]interface{}, kuberneteslbs model.LabelSet) er
 	return nil
 }
 
+var tagKey = "tag"
+var tagPrefix = "kubernetes.var.log.containers"
+var tagExpression = ".(.*)_(.*)_(.*)-.*.log"
+
+func extractKubernetesMetadataFromTag(records map[string]interface{}) error {
+	tag, ok := records[tagKey].(string)
+	if !ok {
+		return errors.New("the tag entry is missing, no kubernetes metadata will be added")
+	}
+
+	re := regexp.MustCompile(tagPrefix + tagExpression)
+	kubernetesMetaData := re.FindStringSubmatch(tag)
+	if len(kubernetesMetaData) != 3 {
+		return fmt.Errorf("invalid format for tag %v. The tag should be in format: %s%s", tag, tagPrefix, tagExpression)
+	}
+
+	records["kubernetes"] = map[string]interface{}{
+		"pod_name":       kubernetesMetaData[0],
+		"namespace":      kubernetesMetaData[1],
+		"container_name": kubernetesMetaData[2],
```

Comment:
This does not seem to be exactly the container name, I think.
I am looking into 
```shell
kube-proxy-5jgm7_kube-system_kube-proxy-b2b5e199a4fc13f653c3dea8a23ba73044388c42ca63140c575f7f6d4a3d54a4.log
```
as an example where  `kube-proxy-5jgm7` is the pod name and `kube-system` is the namespace.

Now here are the container name and ID:
```shell
$ docker ps --no-trunc
CONTAINER ID                                                       IMAGE                                                                     COMMAND                                                                             CREATED             STATUS              PORTS               NAMES
b2b5e199a4fc13f653c3dea8a23ba73044388c42ca63140c575f7f6d4a3d54a4   sha256:4cd78060c7c4ee77617f2e1dd7c470b12e1719b0f45e6a3626173fb9a4be55d4   "/usr/local/bin/kube-proxy --config=/var/lib/kube-proxy-config/config.yaml --v=2"   5 hours ago         Up 5 hours                              k8s_kube-proxy_kube-proxy-5jgm7_kube-system_a6876e80-4343-4abe-86de-05098ad166ad_0
```

I guess here the container name must be `kube-proxy` only, isn't it?

Reply:
The regex which is used: `var tagExpression = ".(.*)_(.*)_(.*)-.*.log"`
will cut the last `-<hash>.log` and will match only the container_name

Reply:
Nice, thanks!.